web/pages/group-invite/[inviteid].tsx

```diff
@@ -0,0 +1,153 @@
+import { GroupInvite } from 'common/group-invite'
+import { getInvite } from 'common/supabase/group-invites'
+import { getGroup } from 'common/supabase/groups'
+import dayjs from 'dayjs'
+import Lottie from 'react-lottie'
+import { Col } from 'web/components/layout/col'
+import { Page } from 'web/components/layout/page'
+import { useIsAuthorized, useUser } from 'web/hooks/use-user'
+import { initSupabaseAdmin } from 'web/lib/supabase/admin-db'
+import { fromNow } from 'web/lib/util/time'
+import * as invitation from '../../public/lottie/invitation.json'
+import { Button } from 'web/components/buttons/button'
+import { firebaseLogin } from 'web/lib/firebase/users'
+import {
+  getUserIsGroupMember,
+  joinGroupThroughInvite,
+} from 'web/lib/firebase/api'
+import { useIsGroupMember } from 'web/hooks/use-group-supabase'
+import { Row } from 'web/components/layout/row'
+import clsx from 'clsx'
+import { useState } from 'react'
+import { useRouter } from 'next/router'
+
+export async function getStaticProps(props: { params: { inviteid: string } }) {
+  const { inviteid } = props.params
+  const adminDb = await initSupabaseAdmin()
+  const invite: GroupInvite = (await getInvite(inviteid, adminDb)) || null
+  const group = invite ? await getGroup(invite.group_id, adminDb) : null
+  return { props: { invite, groupName: group?.name, groupSlug: group?.slug } }
```

Comment:
Might want to add revalidate: 60s so that it more closely tracks is_max_uses_reached.

That would work except for the last guy, because it always sends even old data immediately, and revalidates for the next page load. So to be more correct, you could change to `getServerSideProps`